testing_app/README.md

```diff
@@ -0,0 +1,52 @@
+# testing_app
+
+A Sample app that shows different types of testing in Flutter.
+
+This particular sample uses the [Provider][] package but any other state management approach
+would do.
+
+[provider]: https://pub.dev/packages/provider
+
+## Goals for this sample
+
+Show how to perform:
+
+- Widget Testing,
+- Flutter Driver(Integration) Testing,
+- Performance Testing, and
+- State Management Testing using the [Provider][] package.
+
+## How to run tests
+- Navigate to the project's root folder using command line and follow the instructions below.
+
+### To run tests using only the Flutter SDK:
+The Flutter SDK can run some of the testing types on its own viz. unit testing and widget testing without the need of a physical device/emulator.
+- To run all the test files in the `test/` directory in one go, run `flutter test`.
+- To run a particular test file, run `flutter test test/<file_path>`
+
+### To run tests on a physical device/emulator:
+- Widget Tests:
+  - Run `flutter run test/<file_path>`
+- Flutter Driver Tests:
```

Comment:
I see in `perf_test.dart` you are tracing timeline. It might make sense to add a sentence here about `--trace-startup` option. This option avoid flushing older timeline events when the timeline gets long, which results in missing of part of the timeline. This options is always on in our device lab tests.

Reply:
Done.